pkg/controllers/state/statenode.go

```diff
@@ -517,3 +518,31 @@ func RequireNoScheduleTaint(ctx context.Context, kubeClient client.Client, addTa
 	}
 	return multiErr
 }
+
+// ClearNodeClaimsCondition will remove the conditionType from the NodeClaim status of the provided statenodes
+func ClearNodeClaimsCondition(ctx context.Context, kubeClient client.Client, conditionType string, nodes ...*StateNode) (err error, requeue bool) {
+	return multierr.Combine(lo.Map(nodes, func(s *StateNode, _ int) error {
+		if !s.Initialized() || s.NodeClaim == nil {
```

Comment:
initialized should capture if the nodeclaim isn't set
```suggestion
		if !s.Initialized() {
```
https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/state/statenode.go#L320

Reply:
Removing this check results in test panics - I don't think your assertions is correct. Here is `Initialized()`:
```
func (in *StateNode) Initialized() bool {
	// Node is managed by Karpenter, so we can check for the Initialized label
	if in.Managed() {
		return in.Node != nil && in.Node.Labels[v1.NodeInitializedLabelKey] == "true"
	}
	// Nodes not managed by Karpenter are always considered Initialized
	return true
}
```
`in.Managed()` return `false` because `in.NodeClaim` is nil. So initialized actually returns `true` when the nodeclaim is `nil`.

Reply:
Most of the time, the nodeclaim should exist if the node has the initialized label, but it's definitely possible for it to go away.